File: conversions/csharp_unity/Location.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Location : BaseElement
{
    // Locality
    public string Scene;
    public string Activity;
    public int? FoundingDate;
    public int? PopulationSize;
    public string ParentLocationId;
    public List<string> PopulationsIds; = new List<string>();
    // Culture
    public string Traditions;
    public string Celebrations;
    public string PrimaryCultId;
    public List<string> SecondaryCultsIds; = new List<string>();
    public List<string> DelicaciesIds; = new List<string>();
    // World
    public string Coordinates;
    public List<string> FoundersIds; = new List<string>();
    // Construction
    public string Logistics;
    public string Architecture;
    public int? ConstructionRate;
    public List<string> BuildingsIds; = new List<string>();
    public List<string> BuildingExpertiseIds; = new List<string>();
    // Production
    public string Extraction;
    public string Industry;
    public int? ExtractionOutput;
    public int? IndustryOutput;
    public string PrimaryResourceId;
    public string PrimaryIndustryId;
    public List<string> SecondaryResourcesIds; = new List<string>();
    public List<string> SecondaryIndustriesIds; = new List<string>();
    // Commerce
    public string Trade;
    public string Infrastructure;
    public string Currency;
    public string PrimaryExtractionMarketId;
    public string PrimaryIndustryMarketId;
    public List<string> SecondaryExtractionMarketsIds; = new List<string>();
    public List<string> SecondaryIndustryMarketsIds; = new List<string>();
    // LocalPolitics
    public string Government;
    public string Opposition;
    public string GoverningTitleId;
    public string PrimaryFactionId;
    public List<string> SecondaryFactionsIds; = new List<string>();
    // RegionalPolitics
    public string TerritorialPolicies;
    public string TerritoryId;
    public string RivalId;
    public string FriendId;
    public List<string> SoftInfluenceOnIds; = new List<string>();
    public List<string> HardInfluenceOnIds; = new List<string>();
    // Strategics
    public string Defensibility;
    public int? Height;
    public string PrimaryFighterId;
    public List<string> SecondaryFightersIds; = new List<string>();
    public List<string> DefensesIds; = new List<string>();
    public List<string> FortificationsIds; = new List<string>();
}

File: languages/typescript/Location.ts
import { BaseElement } from './base_elements';

export interface Location extends BaseElement {
  // Locality
  scene?: string | null;
  activity?: string | null;
  foundingDate?: number | null;
  populationSize?: number | null;
  parentLocationId?: string | null;
  populationsIds?: string[] | null;
  // Culture
  traditions?: string | null;
  celebrations?: string | null;
  primaryCultId?: string | null;
  secondaryCultsIds?: string[] | null;
  delicaciesIds?: string[] | null;
  // World
  coordinates?: string | null;
  foundersIds?: string[] | null;
  // Construction
  logistics?: string | null;
  architecture?: string | null;
  constructionRate?: number | null;
  buildingsIds?: string[] | null;
  buildingExpertiseIds?: string[] | null;
  // Production
  extraction?: string | null;
  industry?: string | null;
  extractionOutput?: number | null;
  industryOutput?: number | null;
  primaryResourceId?: string | null;
  primaryIndustryId?: string | null;
  secondaryResourcesIds?: string[] | null;
  secondaryIndustriesIds?: string[] | null;
  // Commerce
  trade?: string | null;
  infrastructure?: string | null;
  currency?: string | null;
  primaryExtractionMarketId?: string | null;
  primaryIndustryMarketId?: string | null;
  secondaryExtractionMarketsIds?: string[] | null;
  secondaryIndustryMarketsIds?: string[] | null;
  // LocalPolitics
  government?: string | null;
  opposition?: string | null;
  governingTitleId?: string | null;
  primaryFactionId?: string | null;
  secondaryFactionsIds?: string[] | null;
  // RegionalPolitics
  territorialPolicies?: string | null;
  territoryId?: string | null;
  rivalId?: string | null;
  friendId?: string | null;
  softInfluenceOnIds?: string[] | null;
  hardInfluenceOnIds?: string[] | null;
  // Strategics
  defensibility?: string | null;
  height?: number | null;
  primaryFighterId?: string | null;
  secondaryFightersIds?: string[] | null;
  defensesIds?: string[] | null;
  fortificationsIds?: string[] | null;
}
